packages/flutter_tools/test/resident_runner_test.dart

```diff
@@ -0,0 +1,83 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'package:flutter_tools/src/device.dart';
+import 'package:flutter_tools/src/resident_runner.dart';
+import 'package:mockito/mockito.dart';
+import 'package:test/test.dart';
+
+import 'src/context.dart';
+
+class TestRunner extends ResidentRunner {
+  bool hasHelpBeenPrinted = false;
+  String receivedCommand;
+
+  TestRunner(List<FlutterDevice> devices)
+      : super(devices);
+
+  @override
+  Future<Null> cleanupAfterSignal() => null;
+
+  @override
+  Future<Null> cleanupAtFinish() => null;
+
+  @override
+  Future<Null> handleTerminalCommand(String code) async {
+    receivedCommand = code;
+  }
+
+  @override
+  void printHelp({bool details}) {
+    hasHelpBeenPrinted = true;
+  }
+
+  @override
+  Future<int> run({Completer<DebugConnectionInfo> connectionInfoCompleter,
+    Completer<dynamic> appStartedCompleter,
+    String route,
+    bool shouldBuild: true}) => null;
+}
+
+void main() {
+  TestRunner testRunner;
+
+  setUp(() {
+    testRunner =
+    new TestRunner(<FlutterDevice>[new FlutterDevice(new MockDevice())]);
```

Comment:
nit: either indent the second line, or, preferred, put the line breaks after the first ( and before the last ) instead of after the =.